plugins/config/lib/configService.js

```diff
@@ -767,6 +775,36 @@ function overrideJsonProperties(originalObject, overrideObject) {
   return overrideObject;
 }
 
+function mergeJsonProperties(originalObject, overrideObject) {
+  var overrideTable = {};
+  var property = null;
+  var keyArray = Object.keys(overrideObject);
+  var key = null;
+  for (let i = 0; i < keyArray.length; i++) {
+    key = keyArray[i];
+    property = overrideObject[key];
+    htPut(overrideTable,key,property);
+  }
+
+  var overrideProperty = null;
+  keyArray = Object.keys(originalObject);
+  for (let i = 0; i < keyArray.length; i++) {
+    key = keyArray[i];
+    property = originalObject[key];
+    overrideProperty = htGet(overrideTable,key);
+    if (!overrideProperty) {
+      overrideObject[key] = property;
+    } else {
+      if (Array.isArray(overrideProperty)) {
+        if (Array.isArray(property)) {
+          Array.prototype.push.apply(overrideProperty, property);
```

Comment:
Even if this works, `overrideProperty.push.apply(overrideProperty,property)` is a little bit more readable to me. IMO.

Reply:
Also, maybe print a debug-level warning if one is an array and one isnt... because something weird happened.